code/obj/item/device/pda2/bot_control.dm

```diff
@@ -169,92 +153,96 @@
 							self_text("[active] ordered to lockdown [guardthis].")
 						else
-							post_status("bot_control", "command", "guard", "active", active, "target", guardthis)
+							post_status("bot_control", "command", "guard", "address_1", active.net_id, "target", guardthis)
 							self_text("[active] ordered to guard [guardthis].")
-						post_status("bot_control", "command", "bot_status", "active", active)
+						post_status("bot_control", "command", "bot_status", "address_1", active.net_id)
 					else
 						src.all_guard = 0
 						if (!botlist.len)
 							PDA.updateSelfDialog()
 							return
+						var/stored_lockdown = src.lockdown
 						SPAWN(0)
-							// yeah its cheating, but holy heck is it laggy to send a zillion signals via PDA
+							// we are gonna try to do this for real, but with a sleep - mylie
 							var/list/bots = list()
-							for(var/obj/machinery/bot/secbot/bot in src.botlist)
-								bot.guard_the_area(guardthis, src.lockdown)
-								bots[bot] = 1
+							for(var/mob/living/critter/robotic/securitron/bot in src.botlist)
+								if(stored_lockdown)
+									post_status("bot_control", "command", "lockdown", "address_1", bot.net_id, "target", guardthis)
+									self_text("[bot] ordered to lockdown [guardthis].")
+								else
+									post_status("bot_control", "command", "guard", "address_1", bot.net_id, "target", guardthis)
+									self_text("[bot] ordered to guard [guardthis].")
+								sleep(2 DECI SECONDS)
 							if(length(bots) >= 1)
 								self_text("[english_list(bots)] ordered to guard [guardthis].")
 
 			if("stop", "go")
-				post_status("bot_control", "command", href_list["op"], "active", active)
-				post_status("bot_control", "command", "bot_status", "active", active)
+				post_status("bot_control", "command", href_list["op"], "address_1", active.net_id)
+				post_status("bot_control", "command", "bot_status", "address_1", active.net_id)
 				if(href_list["op"] == "go")
 					self_text("[active] set to patrol.")
 				else
 					self_text("[active] set to not patrol.")
 
-			if("summon")
-				post_status("bot_control", "command", "summon", "active", active, "target", summon_turf )
-				post_status("bot_control", "command", "bot_status", "active", active)
+			if("summon") // not spoofable
+				post_status("bot_control", "command", "summon", "address_1", active.net_id, "target", summon_turf)
+				post_status("bot_control", "command", "bot_status", "address_1", active.net_id)
 				self_text("[active] summoned to [summon_turf.loc].")
 
-			if("proc")
-				post_status("bot_control", "command", "proc", "active", active)
-				post_status("bot_control", "command", "bot_status", "active", active)
-				self_text("[active] action request sent.")
-
-			if("summonall")
+			if("summonall") // also not spoofable
 				if (!botlist.len)
 					PDA.updateSelfDialog()
 					return
 				SPAWN(0)
+					// trying this for real - mylie
```

Comment:
?

Reply:
This used to be packetless, did not notice any performance issues with this version